esp-hal-common/src/system.rs

```diff
@@ -276,8 +338,20 @@ pub struct CpuControl {
 pub struct Dma {
     _private: (),
 }
+#[cfg(not(any(esp32c6, esp32)))]
 
 /// The SYSTEM/DPORT splitted into it's different logical parts.
+pub struct SystemParts<'d> {
+    _private: PeripheralRef<'d, SystemPeripheral>,
+    pub software_interrupt_control: SoftwareInterruptControl,
+    pub peripheral_clock_control: PeripheralClockControl,
+    pub clock_control: SystemClockControl,
+    pub cpu_control: CpuControl,
+    #[cfg(pdma)]
+    pub dma: Dma,
+}
```

Comment:
There only seems to be a difference of one field here, probably best not to duplicate the structure if not needed and just `cfg` out the one field.

Reply:
Changed this now